### PR TITLE
Bug 1353784 - Update core schema with optional campaign field

### PR DESCRIPTION
**Pull request by fbertsch**
Also added the parquetmr format.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla-services/mozilla-pipeline-schemas/41)

### CODE REVIEW

File: telemetry/core.parquetmr.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    required binary device (UTF8);
+    required binary arch (UTF8);
+    optional int64 profileDate;
+    optional binary defaultSearch (UTF8);
+    optional binary distributionId (UTF8);
+    optional binary created (UTF8);
+    optional int64 tz;
+    optional int64 sessions;
+    optional int64 durations;
+    optional group searches {
```
Comment (trink) [inline telemetry/core.parquetmr.txt]: Empty groups are not allowed

File: telemetry/core.parquetmr.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+    optional binary experiments (UTF8);
```
Comment (trink) [inline telemetry/core.parquetmr.txt]: experiments is an array

**Comment by fbertsch** [comment]
@trink RFAL

File: telemetry/core.parquetmr.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    optional int64 durations;
+    optional group searches (MAP) {
+        repeated group key_value {
+            required binary key (UTF8);
+            required binary value (UTF8);
+        }
+    }
+    optional group experiments (LIST) {
+        repeated group list {
+            required binary element;
```
Comment (trink) [inline telemetry/core.parquetmr.txt]: You probably want this to be (UTF8)

**Comment by fbertsch** [comment]
@trink good?

File: telemetry/core.parquetmr.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    required binary device (UTF8);
+    required binary arch (UTF8);
+    optional int64 profileDate;
+    optional binary defaultSearch (UTF8);
+    optional binary distributionId (UTF8);
+    optional binary created (UTF8);
+    optional int64 tz;
+    optional int64 sessions;
+    optional int64 durations;
+    optional group searches (MAP) {
```
Comment (trink) [inline telemetry/core.parquetmr.txt]: Value type mismatch with the json schema

File: telemetry/core.schema.json [hunk 5]
```diff
@@ -52,7 +52,10 @@
       "type" : "integer"
     },
     "searches" : {
```
Comment (trink) [inline telemetry/core.schema.json]: It doesn't look like we allow an empty object so "minProperties": 1 is advisable

File: telemetry/core.schema.json [hunk 6]
```diff
@@ -52,7 +52,10 @@
...
-      "type" : "object"
+      "type" : "object",
+      "additionalProperties": {
+        "type": "integer"
+      }
     },
     "experiments" : {
       "type" : "array",
```
Comment (trink) [inline telemetry/core.schema.json]: It doesn't look like we allow an empty array so "minItems": 1 is advisable

File: telemetry/core.parquetmr.txt (hunk 4 above)
Comment (fbertsch) [inline telemetry/core.parquetmr.txt]: Fixed.

File: telemetry/core.schema.json (hunk 5 above)
Comment (fbertsch) [inline telemetry/core.schema.json]: Going to leave this as-is, downstream consumers should be fine.

File: telemetry/core.schema.json (hunk 6 above)
Comment (fbertsch) [inline telemetry/core.schema.json]: Going to leave this as-is, downstream consumers should be fine.

**Pull request merged**